appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/BlocklyPanel.java

```diff
@@ -71,7 +71,8 @@
     exportMethodsToJavascript();
     // Tell the blockly world about companion versions.
     setLanguageVersion(YaVersion.YOUNG_ANDROID_VERSION, YaVersion.BLOCKS_LANGUAGE_VERSION);
-    setPreferredCompanion(YaVersion.PREFERRED_COMPANION, YaVersion.COMPANION_UPDATE_URL,
+    setPreferredCompanion(MESSAGES.useCompanion(BlocklyPanel.getCompVersion(), BlocklyPanel.getCompVersion() + "u"),
```

Comment:
Note that you still need to use `YaVersion.PREFERRED_COMPANION` here, not `BlocklyPanel.getCompVersion()`, as this line sets the companion version so calling the getCompVersion method returns undefined.